lib/manager.rb

```diff
@@ -0,0 +1,114 @@
+require "pry"
+require_relative "room"
+
+module Hotel
+  class Manager
+    attr_accessor :rooms_reservations_hash
+
+    def initialize
+      @rooms_reservations_hash = Room.make_rooms
+    end
+
+    # Manager.rooms_res_hash builds this structure:
+    # {
+    #     1:[],
+    #     2:[],
+    #     3:[],
+    #     4:[]
+    #      ...
+    #     20:[]
+    # }
+
+    def list_rooms
+      all_room_numbers = @rooms_reservations_hash.keys.sort!
+      return all_room_numbers
+    end
+
+    def list_reservations_for_date(date)
+      res_on_date = []
+      @rooms_reservations_hash.each do |room, reservations|
+        reservations.each do |res|
+          if res.date_range_string_array.include? (date.to_s)
+            this_rooms_reservations = [room]
+            this_rooms_reservations << (res.date_range_string_array.first + " through " + res.date_range_string_array.last)
+          end
+          res_on_date << this_rooms_reservations unless this_rooms_reservations == nil
+        end
+      end
+
+      return res_on_date # array with [[1, "2019-06-10 through 2019-06-12"], [room_num, "date through date"], etc]
```

Comment:
This works, but I'd suggest that it would be easier as an array of hashes with the keys `room` and `reservation_date`, just for readability.